### DISCUSSION TITLE
Render at native resolution from stable to lazer.

### DISCUSSION THREAD

**Discussion by eroneroneron** [ppy/osu#16828] [Ideas]
Display scaling is different from Render at native resolution from stable and messes up my sens because i was used to 1600x900 and i thought that maybe adding "render at native resolution" on lazer can fix it so other people don't need to fix their sens.

**Comment by integerrr** [comment]
Display scaling shouldn't "mess up your sensitivity" as it isn't changing your display resolution. 
If you are asking for the letterbox option when choosing a different game resolution, I'm not sure how feasible it would be to implement into lazer and I'll leave that for the dev team to answer, but for now I get around this by calculating and setting display scaling to roughly match the letterboxed resolution

**Reply by frenzibyte** [reply]
osu!lazer's "Display scaling" setting does have a different behaviour to osu!stable's "Render at native resolution" in regards to mouse sensitivty, in a way that osu!lazer's "Display scaling" setting does not affect the mouse area, while osu!stable does and therefore the sensitivty feels lower/higher depends on the selected resolution.

**Reply by integerrr** [reply]
Hm interesting, I stand corrected then, I was under the impression that stable letterbox also doesn't change mouse area. I did wonder why the mouse sensitivity felt slightly different but I put that up to different resolution of the new device. 

Anyway, is it possible to implement this feature? If not then is something like an aspect ratio lock/manual input for the scaling possible down the line?

**Comment by Moltanicaa** [comment]
Stable somehow with native rendering completely changes the interaction of the tablet and the area with it in such a way that at the original 1920x1080 (on the monitor) and 1600x900 (in the osu!stable + native rendering), the cursor does not move beyond the visible part of the screen, but at the same time 100% the tablet area widths are traversed in 900 pixels (not 1080). 

At this time, the lazer ignores the specified scale settings and allows you to freely go beyond the visible boundaries of the playing field. At the same time, if you drag the tablet pen from edge to edge, then the tablet will pass 1080 pixels (100% of the screen), and not 900 pixels (as in stable + native rendering). This greatly spoils the gaming experience when playing at a reduced screen resolution.

The main idea of the request is that it is necessary:
1. Ability to limit the position of the cursor in the visible area (during the game)
2. Consider the possibility of implementing native rendering for similarity, as done in the stable. Yes, I also think that it happened by accident, but it's very cool.

**Reply by peppy** [reply]
That just sounds like you have raw input turned on in stable. I'm very unsure whether we would want to do the same in lazer, it's very weird behaviour (and was only done how it was in stable due to implementation limitations).

**Reply by Moltanicaa** [reply]
It will be enough to introduce a cursor limit in the visible area during the game.
